src/Helper/Quote.php

```diff
@@ -109,16 +109,42 @@ public function __construct(
     public function getValidatedCoupons()
     {
         $coupons = [];
+        $returnInvalidCartRuleCoupon = false;
+        $amount = $this->quote->getSubtotal() - $this->quote->getSubtotalWithDiscount();
```

Comment:
maybe needs to be renamed this `amount` variable, amount of what? Maybe someone who does not know mage will not understand that this is discount. 
P.S. yeah, I know this is my sloppy code copied over

Reply:
k, $discountAmount it is